jungminju/사다리 타기-2469/Main.java

```diff
@@ -0,0 +1,104 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Main {
+    static int row;
+    static int column;
+    static int K;
+    static Map<Integer, Integer> inOrder = new HashMap<>();
+    static Character ladder[][];
```

Comment:
해당 코드에서는 굳이 Character 객체 배열로 선언할 필요가 없을 것 같습니다.
char[][] 배열로 변경하면 메모리 사용을 줄일 수 있을 것 같습니다.
Character 객체 배열과 같이 객체 배열을 사용하면 각 요소가 객체 헤더로 인해 대략 12 바이트 정도의 오버헤드가 생긴다고 합니다.
해당 코드에서는 큰 문제가 되지 않겠지만 나중에 더 큰 배열을 다루게 된다면 char과 같은 Primitive 타입을 사용하는 것이 더 효율적일 것 같습니다.

```suggestion
    static char ladder[][];
```

Reply:
객체 헤더로 인해 오버헤드가 생기는 건 미처 몰랐습니다! 좋은 정보 감사합니다.